src/Interpreters/Streaming/Aggregator.h

```diff
@@ -371,6 +371,17 @@ struct AggregatedDataVariants : private boost::noncopyable
         /// proton: ends;
     }
 
+    void reset();
+
+    void resetAggregatesPool()
+    {
+        aggregates_pools = Arenas(1, std::make_shared<Arena>());
+        aggregates_pool = aggregates_pools.back().get();
+        aggregates_pool->enableRecycle(true);
```

Comment:
For global aggregation, do we need enable recycle ?

Reply:
Let's use `initAggregatesPool()` or `resetAndCreateAggregatesPools()`

Reply:
> For global aggregation, do we need enable recycle ?

it will be disable in `AggregatedDataVariants::init()`

Reply:
Let's put a comment there 

```
/// Enable GC for arena by default. For cases like global aggregation, we will disable it further in \init
```